src/formatDate.js

```diff
@@ -8,7 +8,42 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let dateArr = [];
+  let resultString = '';
+  let index = 0;
+
+  dateArr = date.split(fromFormat[3]);
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    const fromIndex = fromFormat.indexOf(toFormat[i]);
+
+    index++;
+
+    if (toFormat[i] === 'YY' && fromFormat.includes('YYYY')) {
+      const number = dateArr[index - 1];
+
+      resultString += number.slice(-2);
+    } else if (toFormat[i] === 'YYYY' && fromFormat.includes('YY')) {
+      const year = Number(dateArr[index - 1]);
+
+      if (dateArr[index - 1] === '00') {
+        resultString += '2000.';
+        continue;
+      } else {
+        const fullYear = year < 30 ? `20${year}` : `19${year}`;
+
+        resultString += fullYear;
+      }
+    } else {
+      resultString += dateArr[fromIndex];
+    }
+
+    if (i < toFormat.length - 2) {
+      resultString += toFormat[3];
+    }
```

Comment:
Here you're adding the separator to `resultString`. However, this is done only if the current index is less than the length of `toFormat` array minus 2. This can lead to missing the separator after the last date part. Consider removing this condition and handling the addition of the separator inside the loop for each date part.